### PR TITLE
Enhance LPDU

### PR DESCRIPTION
**Pull request by tobylorenz**
According to spec, LPDU is medium-dependent, so doesn't know how to convert from/to packages. So I shifted this functionality out into Communication Medium files, and then added the remaining LPDUs. If the patch is too large, I can split it into smaller parts.

Signed-off-by: Tobias Lorenz <tobias.lorenz@gmx.net>

### CODE REVIEW

File: src/libserver/busmonitor.cpp [hunk 1]
```diff
@@ -101,10 +101,10 @@ A_Busmonitor::send_L_Busmonitor (LBusmonPtr p)
       buf.resize (7);
       EIBSETTYPE (buf, EIB_BUSMONITOR_PACKET_TS);
       buf[2] = p->status;
-      buf[3] = (p->timestamp >> 24) & 0xff;
-      buf[4] = (p->timestamp >> 16) & 0xff;
-      buf[5] = (p->timestamp >> 8) & 0xff;
-      buf[6] = (p->timestamp) & 0xff;
+      buf[3] = p->time_stamp >> 24;
```
Comment (smurfix) [inline src/libserver/busmonitor.cpp]: Please either don't drop the mask on this line, or remove them all.  Any reasonable compiler will know that *all* of these masking ANDs are no-ops anyway. Leaving it off only on the top line looks strange, i.e. when I read that I'd pause to think whether the omission is in any way intentional.

**Comment by mfechner** [comment]
I suggest using the names that are also used in the KNX specification.
You can get the KNX spec for free here:
https://my.knx.org/en/shop/knx-specifications

Check there Volume 3 Part 3/3:
image

So destination_address matches here the spec which makes the spec together with the code easier readable. Just my opinion ;)

**Comment by smurfix** [comment]
I do have the spec. However I'd say that "dest_addr" is better than "destination_address". I can read the former at one glance while the longer name requires consciously scanning the whole word. At least for me that takes away some mental bandwidth away from actually understanding the code.

There's a balance to be found between "dst" or "dest" which by itself doesn't say whether it's an address or an interface or whatever, and "destination_address" which is just plain too much to read (let alone type).

File: src/libserver/busmonitor.cpp (hunk 1 above)
Comment (tobylorenz) [reply src/libserver/busmonitor.cpp]: Yes, ok. Good point. I've reverted it.

**Comment by tobylorenz** [comment]
Indeed I renamed the variables to be in accordance with the specs. For me it's then easier to hold it side-by-side.

**Comment by tobylorenz** [comment]
I split the patch into four parts now for easier review.

**Comment by smurfix** [comment]
OK, fair enough, you're the one doing the work here.
Please rebase or merge to current master.

**Comment by tobylorenz** [comment]
Rebase done. :-)

**Pull request merged**